### DISCUSSION TITLE
Is setting `listen_on` required to use remote control?

### DISCUSSION THREAD

**Discussion by rivenirvana** [kovidgoyal/kitty#6537] [Q&A] [answered]
Posting this as a follow-up on issue #6535, and just want clarification.

It seems that the cause of my problem was only kind of tangential to the described issue and proposed fix. Apparently, I had `allow_remote_control yes` set in my `kitty.conf` awhile ago and forgot about it, and likely after commit bde4a41 is when I couldn't launch kitty as per #6535. Removing the option made kitty launch again without error, and `allow_remote_control` does not work without also setting `listen_on`.

Is this the new and intended behavior? This is just a bit confusing because as I said, I had `allow_remote_control yes` in my config before and kitty was launching fine despite not using it or setting `listen_on` on invocation. Also, the [tutorial](https://sw.kovidgoyal.net/kitty/remote-control/) in the docs does not state that setting `listen_on` is required to use remote control.

Seeing this [line](https://github.com/kovidgoyal/kitty/blob/master/kitty/boss.py#L363) after following the traceback, does it not seem that in my case, this code path is active when I had `allow_remote_control` set but had never set `listen_on`?
``` python
if args.listen_on and self.allow_remote_control in ('y', 'socket', 'socket-only', 'password'):
```

Any insight towards this issue would be greatly appreciated.

**Comment by kovidgoyal** [comment, marked as answer]
https://github.com/kovidgoyal/kitty/commit/b348195d22f75d47d57e9865ec44b11e046c350b

**Reply by rivenirvana** [reply]
Will check if this fixes it once tests are fixed. Builds are failing because of failed `test_transfer_send (kitty_tests.file_transmission.TestFileTransmission)`

**Reply by rivenirvana** [reply]
Can confirm issue is fixed now, thanks!
